flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/MeasurementInfoProvider.java

```diff
@@ -63,7 +62,6 @@ private static String getScopedName(String metricName, MetricGroup group) {
     }
 
     private static String getLogicalScope(MetricGroup group) {
-        return ((FrontMetricGroup<AbstractMetricGroup<?>>) group)
-                .getLogicalScope(CHARACTER_FILTER, SCOPE_SEPARATOR);
+        return ((LogicalScopeProvider) group).getLogicalScope(CHARACTER_FILTER, SCOPE_SEPARATOR);
```

Comment:
I'd suggest to add a utility that check whether `group` implements `LogicalScopeProvider` and throws a meaningful exception (e.g. `IllegalStateException` with a proper exception message). This utility could be reused at other places.